index.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const semver = require('semver');
+const execa = require('execa');
+const del = require('del');
+
+const exec = (cmd, args) => {
+	const result = execa.sync(cmd, args);
+	console.log(result.stdout);
+	console.log(result.stderr);
+};
+
+module.exports = input => {
+	input = input || 'patch';
+
+	if (['patch', 'minor', 'major'].indexOf(input) === -1 && !semver.valid(input)) {
+		throw new Error('Version should be either path, minor, major, or a valid semver version.');
+	}
+
+	if (semver.gte(process.version, '6.0.0')) {
+		throw new Error('You should not publish when running Node.js 6. Please downgrade and publish again. https://github.com/npm/npm/issues/5082');
+	}
+
+	if (execa.sync('git', ['status', '--porcelain']).stdout !== '') {
+		throw new Error('Unclean working tree. Commit or stash changes first.');
+	}
+
+	if (execa.sync('git', ['rev-list', '--count', '--left-only', '@{u}...HEAD']).stdout !== '0') {
```

Comment:
Does this actually work? I updated `sindre-playground` via a separate clone and ran the command

```
$ git rev-list --count --left-only @'{u}'...HEAD
0
```